packages/block-editor/src/components/block-card/index.js

```diff
@@ -30,29 +30,52 @@ function BlockCard( { title, icon, description, blockType, className } ) {
 	const isOffCanvasNavigationEditorEnabled =
 		window?.__experimentalEnableOffCanvasNavigationEditor === true;
 
-	const { parentNavBlockClientId } = useSelect( ( select ) => {
-		const { getSelectedBlockClientId, getBlockParentsByBlockName } =
-			select( blockEditorStore );
+	const { closestControllingBlock } = useSelect( ( select ) => {
+		const {
+			getSelectedBlockClientId,
+			getBlockParents,
+			areInnerBlocksControlled,
+			getBlock,
+		} = select( blockEditorStore );
 
 		const _selectedBlockClientId = getSelectedBlockClientId();
 
+		const blockParents = getBlockParents( _selectedBlockClientId, true );
+
+		// Find the first parent that is controlling its inner blocks.
+		const closestControllingBlockClientId = blockParents.find(
+			( parentBlockClientId ) =>
+				areInnerBlocksControlled( parentBlockClientId )
+		);
+
+		const _closestControllingBlock = getBlock(
+			closestControllingBlockClientId
+		);
+
 		return {
-			parentNavBlockClientId: getBlockParentsByBlockName(
-				_selectedBlockClientId,
-				'core/navigation',
-				true
-			)[ 0 ],
+			closestControllingBlock: _closestControllingBlock,
 		};
 	}, [] );
 
 	const { selectBlock } = useDispatch( blockEditorStore );
 
+	// If the selected block is being controlled then show a back button which
+	// allows jumping immediately up to the controlling ancestor block.
+	// Currently limited to the offcanvas experiment but may be opened up to all
+	// blocks which meet the criteria in future.
+	const showBackButton =
+		isOffCanvasNavigationEditorEnabled &&
+		closestControllingBlock?.name !== 'core/template-part' &&
+		closestControllingBlock?.name !== 'core/block';
+
 	return (
 		<div className={ classnames( 'block-editor-block-card', className ) }>
-			{ isOffCanvasNavigationEditorEnabled && parentNavBlockClientId && (
+			{ showBackButton && (
 				<Button
-					onClick={ () => selectBlock( parentNavBlockClientId ) }
-					label={ __( 'Go to parent Navigation block' ) }
+					onClick={ () =>
+						selectBlock( closestControllingBlock?.clientId )
+					}
+					label={ __( 'Go to controlling block' ) }
```

Comment:
Please make this label more explanatory. Example: `label={ sprintf( __( 'Back to parent block: %s', parentBlock.title ) ) }`